plugins/woocommerce/client/blocks/assets/js/blocks/product-details/test/block.tsx

```diff
@@ -0,0 +1,202 @@
+import '../../accordion/inner-blocks/accordion-panel';
+import '../../product-description';
+import '../../product-reviews';
+import '../../product-specifications';
+import '../../single-product';
+
+import { initializeEditor } from '../../../../../tests/integration/helpers/integration-test-editor';
+import {
+	productWithoutSpecifications,
+	productWithSpecifications,
+} from '../fixture';
+
+async function setupWithSingleProduct(
+	attributes: BlockAttributes,
+	productId: number
+) {
+	const productDetailsBlock = createBlock(
+		'woocommerce/product-details',
+		attributes
+	);
+
+	const singleProductBlock = [
+		{
+			name: 'woocommerce/single-product',
+			attributes: {
+				productId,
+			},
+			innerBlocks: [ productDetailsBlock ],
+		},
+	];
+
+	return initializeEditor( singleProductBlock );
+}
+
+jest.mock( '@wordpress/data', () => ( {
+	...jest.requireActual( '@wordpress/data' ),
+	useSelect: jest.fn(),
+} ) );
+
+describe( 'Product Details block', () => {
+	describe( 'Single Product block', () => {
+		const server = setupServer(
+			http.get( '/wc-admin/options', ( { request } ) => {
+				const url = new URL( request.url );
+				const options = url.searchParams.get( 'options' );
+				// Check if the request is for dimension and weight units
+				if (
+					options ===
+					'woocommerce_dimension_unit,woocommerce_weight_unit'
+				) {
+					return HttpResponse.json( {
+						woocommerce_dimension_unit: 'cm',
+						woocommerce_weight_unit: 'kg',
+					} );
+				}
+				// Default response for other options requests
+				return HttpResponse.json( {}, { status: 200 } );
+			} ),
+			http.get( '/wc/store/v1/products/*', () => {
+				return HttpResponse.json( productWithSpecifications );
+			} ),
+			http.get( '/wc/v3/products/*', () => {
+				return HttpResponse.json( productWithSpecifications );
+			} ),
+			http.get( '*', () => {
+				return HttpResponse.json( {} );
+			} ),
+			http.options( '*', () => {
+				return HttpResponse.json( {} );
+			} )
+		);
+
+		beforeAll( () => server.listen() );
+
+		beforeEach( () => {
+			( useSelect as jest.Mock ).mockImplementation(
+				( callback, deps ) => {
+					const originalUseSelect =
+						jest.requireActual( '@wordpress/data' ).useSelect;
+					const originalResult = originalUseSelect( callback, deps );
+
+					if (
+						originalResult &&
+						typeof originalResult === 'object' &&
+						! Array.isArray( originalResult )
+					) {
+						const result = {
+							...originalResult,
+							wasBlockJustInserted: true,
+						};
+
+						return result;
+					}
+					return originalResult;
+				}
+			);
+		} );
+
+		afterEach( () => {
+			( useSelect as jest.Mock ).mockClear();
+		} );
+
+		afterAll( () => server.close() );
+
+		test( 'should render product specifications when product is selected', async () => {
+			await setupWithSingleProduct( {}, 2 );
+
+			await waitFor( () => {
+				expect(
+					screen.queryByRole( 'button', { name: /description/i } )
```

Comment:
Why do we use `queryByRole` vs `getByRole`? Usually, the rule of thumb is when we expect the element to be found, we choose `getByRole()`. We use `queryByRole()` when the element may/will not be found.